newrelic/core/transaction_node.py

```diff
@@ -261,7 +268,8 @@ def apdex_metrics(self, stats):
         frustrating = 0
 
         if self.errors:
-            frustrating = 1
+            if False in (error.expected for error in self.errors):
```

Comment:
Let's flatten this to `if self.errors and False in (error.expected for error in self.errors):`. Otherwise the else statement will be missed when an expected error occurs.

Reply:
That's the way I actually had it originally but I started second guessing the logic in the case an error was expected. I'll switch it back!